iml-gui/crate/src/lib.rs

```diff
@@ -565,124 +536,140 @@ fn handle_record_change(
-                    orders
-                        .proxy(Msg::Tree)
-                        .send_msg(tree::Msg::Add(warp_drive::RecordId::Host(id)));
-                };
+                    if model.records.filesystem.insert(id, Arc::clone(&x)).is_none() {
+                        orders
+                            .proxy(Msg::Tree)
+                            .send_msg(tree::Msg::Add(warp_drive::RecordId::Filesystem(id)));
+                    }
 
-                orders.proxy(Msg::ServersPage).send_msg(page::servers::Msg::SetHosts(
-                    model.records.host.values().cloned().collect(),
-                    model.records.lnet_configuration.clone(),
-                    model.records.pacemaker_configuration.clone(),
-                    model.records.corosync_configuration.clone(),
-                ));
-            }
-            warp_drive::Record::ManagedTargetMount(x) => {
-                model.records.managed_target_mount.insert(x.id, Arc::new(x));
-            }
-            warp_drive::Record::OstPool(x) => {
-                model.records.ost_pool.insert(x.id, Arc::new(x));
-            }
-            warp_drive::Record::OstPoolOsts(x) => {
-                let id = x.id;
-                if model.records.ost_pool_osts.insert(x.id, Arc::new(x)).is_none() {
                     orders
-                        .proxy(Msg::Tree)
-                        .send_msg(tree::Msg::Add(warp_drive::RecordId::OstPoolOsts(id)));
-                };
-            }
-            warp_drive::Record::StratagemConfig(x) => {
-                model.records.stratagem_config.insert(x.id, Arc::new(x));
-            }
-            warp_drive::Record::Target(x) => {
-                let id = x.id;
-                let x = Arc::new(x);
+                        .proxy(Msg::Page)
+                        .proxy(page::Msg::Filesystems)
+                        .send_msg(page::filesystems::Msg::AddFilesystem(x));
+                }
+                ArcRecord::ContentType(x) => {
+                    model.records.content_type.insert(x.id, x);
+                }
+                ArcRecord::Group(x) => {
+                    model.records.group.insert(x.id, x);
+                }
+                ArcRecord::Host(x) => {
+                    let id = x.id;
+                    if model.records.host.insert(x.id, x).is_none() {
+                        orders
+                            .proxy(Msg::Tree)
+                            .send_msg(tree::Msg::Add(warp_drive::RecordId::Host(id)));
+                    };
 
-                if model.records.target.insert(x.id, Arc::clone(&x)).is_none() {
                     orders
-                        .proxy(Msg::Tree)
-                        .send_msg(tree::Msg::Add(warp_drive::RecordId::Target(id)));
-                };
+                        .proxy(Msg::Page)
+                        .proxy(page::Msg::Servers)
+                        .send_msg(page::servers::Msg::SetHosts(
```

Comment:
Since we are passing more than hosts, should we call this message something else?

Reply:
The next time this is updated, it will be to use the `RecordChange` trait, so the message will go away.

Reply:
👍